Source/WebCore/page/Navigation.cpp

```diff
@@ -39,6 +40,28 @@ Navigation::Navigation(ScriptExecutionContext* context, LocalDOMWindow& window)
 {
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#initialize-the-navigation-api-entries-for-a-new-document
+void Navigation::initializeEntries(const Ref<HistoryItem>& currentItem, const Vector<Ref<HistoryItem>>& items)
+{
+    for (auto item : items)
+        m_entries.append(NavigationHistoryEntry::create(scriptExecutionContext(), item->url()));
+    m_currentEntryIndex = items.find(currentItem);
+}
+
+Vector<Ref<NavigationHistoryEntry>> Navigation::entries() const
```

Comment:
It is not great to be copying a Vector of refpointers whenever this is called. It's a lot of unnecessary churn. If I were you, I'd store an empty vector in a `static NeverDestroyed<Vector<Ref<NavigationHistoryEntry>>>` and return that in the early return case. This way the return type can be `const Vector<Ref<NavigationHistoryEntry>>`

Reply:
Fair enough, I added it.